lib/restrictions.js

```diff
@@ -50,13 +57,26 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
  * @returns {Promise<object>} An object with the inserted record.
  */
 export async function insert({restriction} = {}) {
+  await validateSchema({restriction});
```

Comment:
We typically don't do this type of validation here but at other levels (e.g., HTTP API). Do we need to deviate from the usual practice here?

Reply:
We are doing the validation here because this package does not have an HTTP API level.

Reply:
I agree, we should validate this restriction on the MRGAS level (which is currently the only thing that's calling this module).

Reply:
@dlongley @dmitrizagidulin So all validation should be removed at this level and moved up to the module that calls this one?

Reply:
@mandyvenables I believe so, yes.